File: src/LifeTrack.Domain/Entities/Transaction/Transaction.cs
namespace LifeTrack.Domain.Entities.Transaction
{
    public class Transaction
    {
        public int Id { get; private set; }
        public string Description { get; private set; }
        public decimal Amount { get; private set; }
        public int CategoryId { get; private set; }
        public int StatusId { get; private set; }
        public DateTime TransactionDate { get; private set; }

        public Transaction(string description, decimal amount, int categoryId, int statusId, DateTime transactionDate)
        {
            Description = description;
            Amount = amount;
            CategoryId = categoryId;
            StatusId = statusId;
            TransactionDate = transactionDate;
        }
    }
}

File: src/LifeTrack.UI.Web/wwwroot/js/Transaction/transaction.js
// Evento de submissão do formulário via AJAX
$('#transactionForm').submit(function (e) {
    e.preventDefault(); // Evita o comportamento padrão de envio do formulário

    let data = {
        description: $('#Description').val(),
        amount: parseFloat($('#Amount').val().replace(",", ".")),
        transactionDate: $('#TransactionDate').val(),
        categoryId: $('#categoryDropdown').val(),
        statusId: 1 // Ajuste conforme necessário
    };

    console.log(data);

    $.ajax({
        url: '/Transaction/Insert',
        type: 'POST',
        contentType: 'application/json',
        data: JSON.stringify(data),
        success: function (response) {
            Swal.fire({
                title: "Sucesso",
                text: response.message,
                icon: "success"
            }).then(() => {
                location.reload(); // recarrega a página após o usuário fechar o alert
            });
        },
        error: function (xhr, status, error) {
            Swal.fire("Erro", "Falha ao cadastrar transação: " + error, "error");
        }
    });
});